2 - RTB/Documentazione esterna/Analisi dei Requisiti/Analisi dei Requisiti.typ

```diff
@@ -237,26 +245,36 @@ $bold("Scenario: ")$
 
 
 
-== UC-4 Creazione scaffale
+= Richiesta di spostamento di un prodotto
+
+#figure(image("./imgs/uc4.png", format: "png"), caption: [UML UC-4])
+
+== Richiesta di spostamento di un prodotto
+
 $bold("Descrizione: ")$ 
-uno scaffale viene creato in base ai valori dati dall'utente e aggiunto nell'ambiente in una posizione valida specificata. Seccessivamente vengono creati i bin contenuti dallo scaffale e posizionati in esso.
+l'utente seleziona il prodotto di cui desidera una ricollocazione all'interno del magazzino e avvia una richiesta di spostamento verso un altro bin.
 
 $bold("Attore: ")$
 utente.
 
 $bold("Precondizioni: ")$
-- l'ambiente di lavoro deve essere stato configurato con successo.
+- devono esistere almeno due bin distinti;
+- uno dei due bin deve contenere un prodotto;
+- uno dei due bin deve essere vuoto.
 
 $bold("Postcondizioni: ")$
-- nell'ambiente di lavoro è stato aggiunto un nuovo scaffale;
-- nello scaffale creato sono stati aggiunti i bin da esso contenuti.
+- il prodotto è contenuto all'interno del bin di destinazione;
```

Comment:
in teoria non è vero che il prodotto spostato da A a B ora si trova in B: infatti il nostro applicativo si limita al momento solamente a dare la possibilità di *richiedere* lo spostamento. A schermo semplicemente verrà mostrata una sorta di notifica/animazione, ma il prodotto rimane sempre nella posizione A. Non si è parlato di aggiornamento della posizione degli elementi a seguito della richiesta di spostamento. Potremmo chiedere maggiori conferme a Bassani la prossima settimana, dato che anche l'utilizzo delle API per comprendere se lo spostamento è andato a buon fine è stato rivisto nella sua ultima mail.

Reply:
hai ragione, so che non possiamo modificare il database infatti intendevo che a seguito dello spostamento il prodotto si trova nel bin di destinazione all'interno del nostro sistema e che con il refresh ritorna dov'era. L'ho espresso in modo ambiguo in effetti, meglio cambiare